src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -49,7 +47,15 @@ data class OpenChannel(
     val channelType: ChannelType.SupportedChannelType
 ) : PeerCommand()
 
-data class BytesReceived(val data: ByteArray) : PeerCommand()
+data class PeerConnection(val id: Long, val output: Channel<LightningMessage>) {
+    fun send(msg: LightningMessage) {
+        // We can safely use trySend because we use unlimited channel buffers.
+        // If the connection was closed, the message will automatically be dropped.
+        output.run { trySend(msg) }
```

Comment:
nit: why the indirection through `run`, it's less readable that way?

```suggestion
        output.trySend(msg)
```

We could also look at the result and log something if the message was dropped, that could be useful when debugging? There are cases where we won't know that a message is dropped (if the TCP connection has been closed but the `output` channel hasn't been closed yet), but it should still be useful.

Reply:
Done in fc4d5fae, with a nice dedicated MDC logger.